client/components/gravatar-caterpillar/index.jsx

```diff
@@ -0,0 +1,62 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { noop, map, size, takeRight, filter, uniqBy } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import Gravatar from 'components/gravatar';
+
+class GravatarCaterpillar extends React.Component {
+	static propTypes = {
+		onClick: PropTypes.func,
+	};
+
+	render() {
+		const { users, onClick, maxGravatarsToDisplay } = this.props;
+
+		if ( size( users ) < 1 ) {
+			return null;
+		}
+
+		const gravatarSmallScreenThreshold = maxGravatarsToDisplay / 2;
+
+		// Only display authors with a gravatar, and only display each author once
+		const displayedUsers = takeRight(
```

Comment:
Why `takeRight` instead of `take`?

Reply:
We generally want to show the most recent avatars first.

For example, in Conversations we pull the `author` key off a chronological list of comments.